### PR TITLE
feat(mod): allow user pass args

### PR DESCRIPTION
**Pull request by j2gg0s**
pre-commit allow user pass arguments to hook, and we need this feature now.
Ref: https://pre-commit.com/#pre-commit-configyaml---hooks

### CODE REVIEW

**Comment by dnephin** [comment]
Thank you for the PR!  I looked at the pre-commit docs, but I don't understand how passing args to hooks works. How do you pass args to this hook?

**Comment by j2gg0s** [comment]
In `pre-commit`, user can pass parameters to hook by `args` .
Doc: https://pre-commit.com/#config-args
Example: https://pre-commit.com/#passing-arguments-to-hooks

pre-commit will execute our bash file and pass args.
All we need to do is pass this args to `go mod tidy`.

If we run go1.17, we can test this by pass argument `-compat 1.17` and find go.sum is same as `go mod tidy -compat 1.17`
```
  - repo: https://github.com/j2gg0s/pre-commit-golang
    rev: d2af21eee9535c84ab242d2b940aa6f03540a276
    hooks:
      - id: go-mod-tidy
        exclude: ^(vendor|third_party)/
        args: [-compat=1.17]
```

File: run-go-mod-tidy.sh [hunk 1]
```diff
@@ -1,13 +1,12 @@
 #!/usr/bin/env bash
 
-if go mod tidy -v 2>&1 | grep -q 'updates to go.mod needed'; then
-    exit 1
+go mod tidy -v $@
+if [ $? -ne 0 ]; then
```
Comment (dnephin) [inline run-go-mod-tidy.sh:4]: Looking at https://github.com/golang/go/blob/master/src/cmd/go/internal/modcmd/tidy.go, it seems like `go mod tidy` does not exist with anything except for `0`.  

I think we still need `| grep -q '...'` to check if `go mod tidy` found problems or not

**Review by dnephin — COMMENTED** [review]
Thanks! I understand now. I think this looks good, just one comment below

File: run-go-mod-tidy.sh (hunk 1 above)
Comment (j2gg0s) [reply run-go-mod-tidy.sh:4]: I'm not familiar with these code, but after a simple browse, `go mod tidy` exit by panic or os.Exit in functions?

In order to verify, we can change any existing dependency to error address, such `github.com/j2gg0s/otsql @xxx` -> `github.com/j2gg1s/otsql @xxx`, and run `go mod tidy`.

Comment (dnephin) [reply run-go-mod-tidy.sh:4]: Hmm, I think you might be right. Seems like maybe that error does result in a `os.Exit` from `Fatalf` here: https://github.com/golang/go/blob/122a22e0e9eba7fe712030d429fc4bcf6f447f5e/src/cmd/go/internal/modload/load.go#L443-L445

Have you been able to test it? I can't seem to get it to print that error.

Comment (j2gg0s) [reply run-go-mod-tidy.sh:4]: I don't known when `go mod tidy` will exit by this code, so use case is difficult to construct.

Answer in this issue also proves that go mod tidy will exit no zero for error.
https://github.com/golang/go/issues/27005

**Review by dnephin — APPROVED** [review]
Ok, let's try it out!

Thank you for the PR!

**Pull request merged**